google-compute-service.js

```diff
@@ -186,6 +180,32 @@ module.exports = class GoogleComputeService {
     }
   }
 
+  async getAddressResourceByIP(searchIP, region, project) {
```

Comment:
I'm sorry, but this function is completely unacceptable.

1. What is it supposed to return? is it a promise? is it an object? The name suggests its the object found by IP. Why is the object returned only in happy path, but all other paths are returning Promise.reject()? It makes no sense.
2. none of the try blocks make any sense. This function should have exactly 0 of those. Not to mention that there is no real scenarios that you should ever use nested try-catch blocks.
3. It's always better to await for the result at the function call, not at the resource iteration.

Please contact me tomorrow so that I can explain how this code should look like.